src/expr/udf/src/external.rs

```diff
@@ -41,25 +46,40 @@ pub struct ArrowFlightUdfClient {
 impl ArrowFlightUdfClient {
     /// Connect to a UDF service.
     pub async fn connect(addr: &str) -> Result<Self> {
-        let conn = tonic::transport::Endpoint::new(addr.to_string())?
-            .timeout(Duration::from_secs(5))
-            .connect_timeout(Duration::from_secs(5))
-            .connect()
-            .await?;
-        let client = FlightServiceClient::new(conn);
-        Ok(Self {
-            client,
-            addr: addr.into(),
-        })
+        Self::connect_inner(
+            addr,
+            ResolutionStrategy::Eager {
+                timeout: TokioDuration::from_secs(5),
+            },
+        )
+        .await
     }
 
-    /// Connect to a UDF service lazily (i.e. only when the first request is sent).
+    /// Connect to a UDF service lazily (i.e. only fetch IP list from DNS when the first request is sent).
     pub fn connect_lazy(addr: &str) -> Result<Self> {
-        let conn = tonic::transport::Endpoint::new(addr.to_string())?
+        block_on(async { Self::connect_inner(addr, ResolutionStrategy::Lazy).await })
```

Comment:
By inspecting the source, it appears that `.channel()` is actually synchronous. We may use `.now_or_never().unwrap()` on the future here.

Reply:
Hmm will it be much higher cost to do `block_on` + `await` here? I prefer not to rely on internal implementations of library.

Reply:
`futures::executor::block_on` is fragile as well. If the future implementation leverages I/O capabilities of `tokio` inside, it will also panic.

Reply:
Updated to `.now_or_never().unwrap()`